sympy/concrete/guess.py

```diff
@@ -377,3 +378,60 @@ def guess_generating_function(v, X=Symbol('x'), types=['all'], maxsqrtn=2):
                 break
 
     return result
+
+
+@public
+def guess(l, stop=False, evaluate=True, niter=None):
+    """
+    This function is adapted from the Rate.m package for Mathematica
+    written by Christian Krattenthaler.
```

Comment:
I don't think; you can see that the package has been adapted for several other CAS (Maxima, Axiom, etc.): http://www.mat.univie.ac.at/%7Ekratt/ See the list in the paragraph:

> If you need to guess sequences of numbers very often, then my Mathematica "guessing machine" RATE (which has now become part of Neil Sloane's Encyclopedia of Integer Sequences) may be useful for you. The Maple implementation by François Béraud and Bruno Gauthier is called GUESS. A Maxima implementation, DEVINE, written by Martin Rubey, is also available. The Axiom guessing package Guess, also written by Martin Rubey, is even more powerful as its range of detected formulas is larger.

But I can't tell much more concerning this.